### ISSUE TITLE
Applying a style to images changes the HR module to clip

### ISSUE THREAD

**Issue by EC1000** [darktable-org/darktable#14050]
**Did you buy darktable from an application store ?**
No

**Describe the bug/issue**
I nearly always apply a style just after importing. I have found that it changes the highlight reconstruction module to "clip highlights" (it used the default "inpaint opposed" before this change). Note that this style does not use the highlight reconstruction module. 
I think it happen only when the image was not opened in the darkroom before applying the style. 

**To Reproduce**
Not totally sure, but I think I found a way to reproduce: 
1. Create a style (I tested by creating a style with just a "local constrast" instance on)
2. Import some raw files (or discard the history from lighttable). Stay in the lighttable and do not go to the darkroom. 
3. Apply the style on _several_ files (the results is not systematic, see below) 
4. Go to darkroom and check the HR module: most of the images have "clip highlights" instead of "inpaint opposed". 

If I only apply the style to one image, it sometimes work correctly (it stays "inpaint opposed"). 
If I apply the style to several images, it sometimes works correctly for the first image only; but the other images are changed. 

**Expected behavior**
HR should keep the "inpaint opposed" reconstruction method. 

**Which commit introduced the error**
I do not know, but it is probably related to inpaint opposed being the new default since 4.2, I suppose. 

**Platform**
* darktable version : both 4.2.1 and master d96c6c5 
* OS                : Linux - kernel 5.15.0-48-generic
* Linux - Distro    : Linux Mint 21.1
* Memory            : 23.3 GB
* Graphics card     : NVIDIA Corporation TU116 [GeForce GTX 1660]
* Graphics driver   : nvidia-driver-510
* OpenCL installed  : yes
* OpenCL activated  : yes
* Xorg              : unknown
* Desktop           : Cinnamon
* GTK+              : unknown
* gcc               : gcc12
* cflags            : unknown
* CMAKE_BUILD_TYPE  : unknown

**Additional context**
 - Can you reproduce with another darktable version(s)? **yes** (4.2.1 and master d96c6c5)
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? **yes**
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? **yes**

**Comment by EC1000** [comment]
Also, I just checked: I reproduce both in append and overwrite modes.

**Comment by MStraeten** [comment]
maybe attach the style so someone can investigate ...

**Comment by EC1000** [comment]
Could it be link to the function to handle the old default (clip) that was not present -- in https://github.com/darktable-org/darktable/pull/12652 by you @TurboGit?

**Comment by MStraeten** [comment]
please provide the style so the devs doesn’t need to guess …

**Comment by EC1000** [comment]
Here is a style (created in master d96c6c5), with which I encounter the problem: [test.dtstyle.txt](https://github.com/darktable-org/darktable/files/11152710/test.dtstyle.txt) (but it is there with any style I use). 

I made some more tests: 
- the condition "images that have not been opened in darkroom" does not seem to be necessary. It was (for me) when I tested in 4.2.1; but with master I can reproduce the problem even if I first open the image in darktable, go back to lighttable and then apply the style.  
- I tested with cr3 files, but also with nef files (to check if it was not cause by the cr3 files / LibRaw instead of rawspeed). I reproduced.
